src-ext/TypeScript/ggvega/src/topLevelSpec.ts

```diff
@@ -125,10 +130,33 @@ function topLevelSpec(ggSpec: GG.TopLevelSpec): VL.TopLevelSpec {
   // not faceted
   topLevelSpec = {
     $schema: vlschema,
-    title: ggSpec.labels.title || undefined,
-    datasets: datasetObject(ggSpec.data),
-    layer: layerArrayByAes(ggSpec.data, ggSpec.layers, ggSpec.scales, ggSpec.labels, ggSpec.coordinates)
+    title: title,
+    datasets: datasets,
+    layer: layer
   };
 
+  // single-view not-faceted
+  if (singleView) {
+
+    if (layer.length > 1) {
+      // warn that we cannot create a single view with more than one layer
+      console.warn('Cannot create single-view spec, returning spec with multiple layers.');
+      return topLevelSpec;
+    }
+
+    // put all of the elements of into single view
+    let topLevelSingleViewSpec: VL.TopLevelSpec = {
+      $schema: vlschema,
+      title: title,
+      datasets: datasets,
+    }
+
+    // append the layer-elements into the top-level spec
+    // Object.assign() is ES6; using this instead: https://stackoverflow.com/a/43675279
+    topLevelSingleViewSpec = {...topLevelSingleViewSpec, ...layer[0]};
```

Comment:
Hi @ijlyttle ,

When I built the `ggvega.js`, I got a warning from `rollup.js`:
 
```
(!) `this` has been rewritten to `undefined`
https://rollupjs.org/guide/en/#error-this-is-undefined
build\ggvega\src\topLevelSpec.js
1: var __assign = (this && this.__assign) || function () {
                   ^
2:     __assign = Object.assign || function(t) {
3:         for (var s, i = 1, n = arguments.length; i < n; i++) {
...and 1 other occurrence
```

When I remove this line, the warning disappears.
```ts
topLevelSingleViewSpec = {...topLevelSingleViewSpec, ...layer[0]};
```

I don't know if it will cause any problems in the future. For now, I think it works well.

Reply:
Hmmm... that line is what is putting the layer spec into the top-level spec, so I don't think I can just remove it....

I think I will need to poke around a bit...

Reply:
I ended up merging the objects explicitly - this seemed to `yarn build` without any warnings and `yarn test` passes.